tests/test_activate.py

```diff
@@ -1105,40 +1107,186 @@ def make_dot_d_files(prefix, extension):
-        pytest.param(("path/number/one", "path/two", "three"), id="tuple with paths"),
+        # falsy
+        pytest.param(None, [None], id="None"),
+        pytest.param("", ["."], id="empty string"),
+        pytest.param((), [()], id="empty tuple"),
+        # native
+        pytest.param(
+            "C:\\path\\to\\One",
+            [
+                "/c/path/to/One",  # MSYS2
+                "/cygdrive/c/path/to/One",  # cygwin
+            ],
+            id="path",
+        ),
+        pytest.param(
+            ["C:\\path\\to\\One"],
+            [
+                ("/c/path/to/One",),  # MSYS2
+                ("/cygdrive/c/path/to/One",),  # cygwin
+            ],
+            id="list[path]",
+        ),
+        pytest.param(
+            ("C:\\path\\to\\One", "C:\\path\\Two", "\\\\mount\\Three"),
+            [
+                ("/c/path/to/One", "/c/path/Two", "//mount/Three"),  # MSYS2
+                (
+                    "/cygdrive/c/path/to/One",
+                    "/cygdrive/c/path/Two",
+                    "//mount/Three",
+                ),  # cygwin
+            ],
+            id="tuple[path, ...]",
+        ),
     ],
 )
-def test_native_path_to_unix(tmp_path: Path, paths: str | Iterable[str] | None):
-    def assert_unix_path(path):
-        return "\\" not in path and ":" not in path
-
-    # convert all path stubs to full paths (localized to current OS)
-    if not paths:
-        pass
-    elif isinstance(paths, str):
-        paths = str(tmp_path / paths)
-    else:
-        paths = tuple(str(tmp_path / path) for path in paths)
-
-    # test that we receive a unix path
-    if isinstance(paths, str) and not paths:
-        assert native_path_to_unix(paths) == "."
-    elif not on_win:
-        assert native_path_to_unix(paths) == paths
-    elif paths is None:
-        assert native_path_to_unix(paths) is None
-    elif isinstance(paths, str):
-        assert assert_unix_path(native_path_to_unix(paths))
-    else:
-        assert all(assert_unix_path(path) for path in native_path_to_unix(paths))
+def test_native_path_to_unix(
+    mocker: MockerFixture,
+    paths: str | Iterable[str] | None,
+    expected: str | list[str] | None,
+) -> None:
+    # test with cygpath
+    assert native_path_to_unix(paths) in expected
+
+    # test without cygpath
+    mocker.patch("subprocess.run", side_effect=FileNotFoundError)
+    assert native_path_to_unix(paths) in expected
+
+
+# Notionally, unix_path_to_native() is only called from
+# _get_path_dirs() with a Unix-style prefix.  However, once the
+# function is known it may get called with all variants.
+#
+# The only tests where *_prefix are used are root filesystem ones.
+windows_prefix = context.target_prefix
+if windows_prefix.startswith("/"):
+    # toggle the prefixes
+    unix_prefix = windows_prefix
+    # yes, this is inception
+    windows_prefix = unix_path_to_native(unix_prefix, unix_prefix)
+else:
+    unix_prefix = native_path_to_unix(windows_prefix)
```

Comment:
@kenodegard I suspect this block (and the commentary) should be in a `if on_win:` clause as much because the tests are in a `skipif (not on_win, ...)` decorator as anything else.